pandas/util/testing.py

```diff
@@ -1049,6 +1051,18 @@ def assert_period_array_equal(left, right, obj='PeriodArray'):
     assert_attr_equal('freq', left, right, obj=obj)
 
 
+def assert_datetime_array_equal(left, right, obj='DatetimeArray'):
+    _check_isinstance(left, right, DatetimeArray)
```

Comment:
why is this needed? why don’t we just have a more generic assert_array_equal

adding more comparison routines generally is just asking for trouble - we already have too many

Reply:
I think we may be able to de-duplicate these once DTA/TDA are EAs, but for now the options are to have this function or to implement something equivalent _inside_ `tm.assert_equal`

Reply:
I think it is fine to reconsider this once all internal ExtensionArrays are in place (I think the same was said when adding the `assert_period_array_equal` that is just above this one)

Reply:
> adding more comparison routines generally is just asking for trouble - we already have too many
 
I really like having separate ones, so that you can assert that you aren't accidentally comparing two of the wrong kind of object. If you just have a generic `assert_array_equal` you could have built two ndarrays on accident and there's no way to check that the type is correct (unless you pass that as a parameter?)

Reply:
again not sure why this is; the point is we are comparing versus an expected value
so this just extra verbose

eg we recently consolidatednto assert_equal to avoid this exact prome
now adding these back is just more code 

sure it’s slightly more explicit but IMHO is not worth the extra functions and maintenance over time

Reply:
> comparing versus an expected value

sometimes though, that expected value isn't constructed explicitly. Sometimes it's the result of some computation (think of the tests in ops for example).

> eg we recently consolidatednto assert_equal to avoid this exact prome
now adding these back is just more code

`assert_equal` just dispatches to each of the specialized asserts. The code has to live somewhere :) But, I don't mean to nitpick over this. I would just rather type

```python
tm.assert_period_array_equal(a, b)
```

than

```python
tm.assert_array_equal(a, b, kind='period')
```

or, worse, `tm.assert_array_equal(a, b)`, because I'm lazy, and accidentally not have a period array.

Reply:
it’s fine

just want to reduce maintenance burden